commcare_connect/utils/ocs_api.py

```
@@ -31,9 +31,12 @@ def list_chatbots(user) -> list[tuple[str, str]]:
             raise OcsApiError(f"Failed to list chatbots: {response.text}")
         except httpx.RequestError as e:
             raise OcsApiError(f"Failed to list chatbots: {e}")
-        data = response.json()
-        chatbots.extend((c["id"], c["name"]) for c in data.get("results", []))
-        url = data.get("next")
+        try:
+            data = response.json()
+            chatbots.extend((c["id"], c["name"]) for c in data["results"])
+            url = data.get("next")
+        except (ValueError, KeyError, TypeError) as e:
+            raise OcsApiError(f"Unexpected chatbots response from OCS: {e}")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)